apps/emqx/src/emqx_trie.erl

```diff
@@ -50,15 +55,20 @@
 -compile(nowarn_export_all).
 -endif.
 
--define(SESSION_TRIE, emqx_session_trie).
 -define(PREFIX(Prefix), {Prefix, 0}).
 -define(TOPIC(Topic), {Topic, 1}).
 
 -record(?TRIE, {
     key :: ?TOPIC(binary()) | ?PREFIX(binary()),
-    count = 0 :: non_neg_integer()
+    count = 0 ::
+        % in case of global mnesia table
+        non_neg_integer()
+        % in case of local ets table
+        | #{_Dest => nil()}
```

Comment:
🤔 seems `nil()` is undefined, and also a bit unclear why we use `[]` as nils (in Erlang 😅)

Reply:
Hm, the [doc](https://www.erlang.org/doc/reference_manual/typespec.html) mentions it, but it might sound confusing indeed. `[]` is also `Nil` in Haskell, so there's probably some heritage line across similar languages. 🌝

Reply:
Wow, really